File: core/src/com/qtech/bubbleblaster/ability/triggers/types/AbilityKeyTriggerType.java
package com.qtech.bubbleblaster.ability.triggers.types;

import com.qtech.bubbleblaster.common.RegistryEntry;
import com.qtech.bubbleblaster.common.init.ObjectInit;

import java.util.Objects;

public class AbilityKeyTriggerType extends RegistryEntry implements ObjectInit<AbilityKeyTriggerType> {
    public static final AbilityKeyTriggerType HOLD = new AbilityKeyTriggerType();

    private final long hash;

    public AbilityKeyTriggerType() {
        this.hash = System.nanoTime();
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        if (!super.equals(o)) return false;
        AbilityKeyTriggerType that = (AbilityKeyTriggerType) o;
        return hash == that.hash;
    }

    @Override
    public int hashCode() {
        return Objects.hash(super.hashCode(), hash);
    }
}

File: src/main/kotlin/qtech/bubbles/ability/triggers/types/AbilityKeyTriggerType.kt
package qtech.bubbles.ability.triggers.types

import qtech.bubbles.common.RegistryEntry
import qtech.bubbles.common.init.ObjectInit
import java.util.*

class AbilityKeyTriggerType : RegistryEntry(), ObjectInit<AbilityKeyTriggerType?> {
    private val hash: Long = System.nanoTime()
    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other == null || javaClass != other.javaClass) return false
        if (!super.equals(other)) return false
        val that = other as AbilityKeyTriggerType
        return hash == that.hash
    }

    override fun hashCode(): Int {
        return Objects.hash(super.hashCode(), hash)
    }

    companion object {
        val HOLD = AbilityKeyTriggerType()
    }
}
